src/entity/service.ts

```diff
@@ -0,0 +1,122 @@
+import fs from 'node:fs/promises'
+import { z } from 'zod'
+import { extendZodWithOpenApi } from '@asteasolutions/zod-to-openapi'
+import partial from 'lodash.partial'
+import { createLogger } from '../logger'
+
+const logger = createLogger(__filename)
+extendZodWithOpenApi(z)
+
+export async function createEntityService(
+  repository: EntityRepository,
+): Promise<EntityService> {
+  return {
+    getAllEntities: partial(getAllEntities, repository),
+    loadEntities: partial(loadEntities, repository),
+  }
+}
+
+export interface EntityService {
+  getAllEntities: () => Promise<Entity[]>
+  loadEntities: () => Promise<void>
+}
+
+export interface EntityRepository {
+  getAllEntities: () => Promise<Entity[]>
+  findById: (id: string) => Promise<Entity | null>
+  addEntity: (entity: Entity) => Promise<Entity>
+  updateEntity: (entity: Entity) => Promise<Entity>
+}
```

Comment:
Maybe `Repository` should be in a separate `types` file?

Reply:
Why do you think so? 🤔

Reply:
Just seems a little out of place in `Service` file, since it is also used in `Repository` file. I understand it is needed in `service` also, but it shares the same name as repository - so I figured a file shared between them may be an ok compromise.
Probably just overthinking it.